cnap/core/rtdb.py

```diff
@@ -77,14 +80,41 @@ def connect(self, host: str = "127.0.0.1", port: int = 6379, db: int = 0) -> Non
         Returns:
             None
         Raises:
-            redis.exceptions.ConnectionError: If connection to the Redis server fails.
+            redis.exceptions.ConnectionError: If connection to the Redis server fails
+                and reconnection exceeds the limit.
         """
         self._conn = redis.Redis(host=host, port=port, db=db)
-        try:
-            self._conn.ping()
-        except redis.exceptions.ConnectionError as e:
-            LOG.error("Failed to connect to Redis")
-            raise e
+
+        sleep_time = 0.5
+
+        # Attempts to reconnect when a connection error occurs, up to MAX_RECONNECTION_TIMES times,
+        # so if a connection error still occurs on the (MAX_RECONNECTION_TIMES + 1)th loop,
+        # raise conncection error
+        for i in range(self.MAX_RECONNECTION_TIMES + 1):
+            try:
+                self._conn.ping()
+            except redis.exceptions.ConnectionError as e:
+                if i == self.MAX_RECONNECTION_TIMES + 1:
```

Comment:
The loop range is defined as range(self.MAX_RECONNECTION_TIMES + 1). The index i will never be equal to self.MAX_RECONNECTION_TIMES + 1?

Reply:
Thanks, fixed.